synapse/api/room_versions.py

```diff
@@ -406,38 +377,13 @@ class RoomVersions:
         msc3083_join_rules=True,
         msc3375_redaction_rules=True,
         msc2403_knocking=True,
-        msc2716_historical=False,
-        msc2716_redactions=False,
         msc3389_relation_redactions=False,
         msc3787_knock_restricted_join_rule=True,
         msc3667_int_only_power_levels=True,
         msc3821_redaction_rules=False,
         msc3931_push_features=(),
         msc3989_redaction_rules=False,
     )
-    MSC2716v4 = RoomVersion(
-        "org.matrix.msc2716v4",
```

Comment:
Removing this room version will mean that users won't be able to leave / interact with such rooms. Not sure if that's a problem for this feature or not

Reply:
ehh, people are going to get left behind regardless of the time period we give and I don't think this was very widely used. I've created https://github.com/matrix-org/synapse/issues/15785 to track a solution to this problem in general.